### PR TITLE
feat(): runs on stable rust

### PR DESCRIPTION
**Pull request by astraw**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,3 +20,8 @@ serde_derive = "~1.0.3"
 default = ["std"]
 
 std = ["serde/std"]
+
+serde-collections = ["serde/collections"]
```
Comment (jrasky) [inline Cargo.toml]: Does it make sense to have all these different features? What about just having one `nightly` feature that does all of those things?

Comment (astraw) [reply Cargo.toml]: Sure, fine to combine all under one feature, but then I would pick `collections` for consistency with serde and to give a hint as to specifically what is needed.

**Comment by jrasky** [comment]
What's the general approach to mutually exclusive features in rust? Cargo has an `--all-features` flag, with which this branch fails to compile:

image

I think the right path here is to default to `std` implementations if both are selected, which is what `serde` does:

image

I'll merge these changes and then add a bit of code to clarify those situations.

**Comment by jrasky** [comment]
Apparently rust-lang/rust#40872 is in the works, so there's a desire for a way to validate feature sets at compile time. In particular, if neither the `std` nor the `collections` feature is selected, then some of the implementation of `Deserializer` gets much less obvious.

I think it makes sense to have a guard for the presence of both `std` and `collections` as features, since that's a situation you can more easily arrive at. Otherwise, leaving the issue open of specifying neither issue, maybe with a note in the documentation or in `Cargo.toml` might be enough until more features land in Rust.

**Comment by jrasky** [comment]
Sorry for all the spam, there were just a couple problem lines in the tests that could just be removed.

**Pull request merged**